source/extensions/filters/http/cache/cacheability_utils.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "common/common/utility.h"
+#include "common/http/headers.h"
+
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+class CacheabilityUtils {
+public:
+  // Checks if a request can be served from cache,
+  // this does not depend on cache-control headers as
+  // request cache-control headers only decide whether
+  // validation is required and whether the response can be cached
+  static bool isCacheableRequest(const Http::RequestHeaderMap& headers);
+
+  // Checks if a response can be stored in cache
+  // Note that if a request is not cacheable according to 'isCacheableRequest'
+  // then its response is also not cacheable
+  // Therefore, 'isCacheableRequest' & 'isCacheableResponse' together
+  // should cover https://httpwg.org/specs/rfc7234.html#response.cacheability
+  static bool isCacheableResponse(const Http::ResponseHeaderMap& headers);
+
+private:
+  const static absl::flat_hash_set<absl::string_view> cacheable_status_codes_;
```

Comment:
Please remove the declaration of cacheaable_status_codes_ from this header file, remove "CacheabilityUtils::" from its definition in the .cc file, put it in an unnamed namespace, and rename it to follow Envoy style rules for non-member constants.

Reply:
Done. 
Just to make sure I understand the principle correctly, any private members/functions that do not use any members of the class are better replaced by variables/functions in an unnamed namespace in the .cc file. Correct?

Reply:
Yes. There are two reasons to put a declaration in a .h file:
1. The language requires it.
2. It's needed by more than one file.

Otherwise, there's no reason to clutter up the header.

Reply:
Thanks for the clarification.